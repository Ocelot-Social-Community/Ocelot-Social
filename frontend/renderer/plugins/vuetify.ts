// eslint-disable-next-line import/no-unassigned-import
import '@mdi/font/css/materialdesignicons.css'
// eslint-disable-next-line import/no-unassigned-import
import 'vuetify/styles'
import { I18n, useI18n } from 'vue-i18n'
import { createVuetify } from 'vuetify'
import { createVueI18nAdapter } from 'vuetify/locale/adapters/vue-i18n'

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export default (i18n: I18n<any, NonNullable<unknown>, NonNullable<unknown>, string, false>) =>
  createVuetify({
    locale: {
      adapter: createVueI18nAdapter({ i18n, useI18n }),
    },
    ssr: true,
  })
