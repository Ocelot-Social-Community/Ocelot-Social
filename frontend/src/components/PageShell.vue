<template>
  <v-app>
    <TopMenu />
    <slot />
  </v-app>
</template>

<script lang="ts" setup>
import TopMenu from '#components/menu/TopMenu.vue'
</script>
