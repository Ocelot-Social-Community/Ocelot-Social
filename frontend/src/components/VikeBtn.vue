<template>
  <v-btn
    :variant="isRouteSelected($attrs.href as string) ? 'tonal' : 'flat'"
    @click.prevent="onClick($attrs.href as string)"
  >
    <slot />
  </v-btn>
</template>
<script lang="ts" setup>
import { navigate } from 'vike/client/router'

import { usePageContext } from '#context/usePageContext'

const pageContext = usePageContext()

function onClick(href: string) {
  return navigate(href)
}

const isRouteSelected = (href: string) => {
  if (href === '/app') {
    return pageContext.urlPathname.indexOf(href) === 0
  }
  return pageContext.urlPathname === href
}
</script>
