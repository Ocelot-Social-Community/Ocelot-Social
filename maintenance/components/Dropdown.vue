<template>
  <v-popover
    v-model:open="isPopoverOpen"
    :open-group="Math.random().toString()"
    :placement="placement"
    :disabled="disabled"
    trigger="manual"
    :offset="offset"
  >
    <slot :toggle-menu="toggleMenu" :open-menu="openMenu" :close-menu="closeMenu" :is-open="isOpen" />
    <template #popover>
<div  @mouseover="popoverMouseEnter" @mouseleave="popoverMouseLeave">
      <slot
        name="popover"
        :toggle-menu="toggleMenu"
        :open-menu="openMenu"
        :close-menu="closeMenu"
        :is-open="isOpen"
      />
    </div>
</template>
  </v-popover>
</template>

<script>
let mouseEnterTimer = null
let mouseLeaveTimer = null

export default {
  props: {
    placement: { type: String, default: 'bottom-end' },
    disabled: { type: Boolean, default: false },
    offset: { type: [String, Number], default: '16' },
  },
  data() {
    return {
      isPopoverOpen: false,
      developerNoAutoClosing: false, // stops automatic closing of menu for developer purposes: default is 'false'
    }
  },
  computed: {
    isOpen() {
      return this.isPopoverOpen
    },
  },
  watch: {
    isPopoverOpen: {
      immediate: true,
      handler(isOpen) {
        try {
          if (isOpen) {
            this.$nextTick(() => {
              setTimeout(() => {
                const paddingRightStyle = `${
                  window.innerWidth - document.documentElement.clientWidth
                }px`
                const navigationElement = document.querySelector('.main-navigation')
                document.body.style.paddingRight = paddingRightStyle
                document.body.classList.add('dropdown-open')
                if (navigationElement) {
                  navigationElement.style.paddingRight = paddingRightStyle
                }
              }, 20)
            })
          } else {
            const navigationElement = document.querySelector('.main-navigation')
            document.body.style.paddingRight = null
            document.body.classList.remove('dropdown-open')
            if (navigationElement) {
              navigationElement.style.paddingRight = null
            }
          }
        } catch (err) {}
      },
    },
  },
  beforeUnmount() {
    clearTimeout(mouseEnterTimer)
    clearTimeout(mouseLeaveTimer)
  },
  methods: {
    toggleMenu() {
      this.isPopoverOpen ? this.closeMenu(false) : this.openMenu(false)
    },
    openMenu(useTimeout) {
      if (this.disabled) {
        return
      }
      this.clearTimeouts()
      if (useTimeout === true) {
        this.popoverMouseEnter()
      } else {
        this.isPopoverOpen = true
      }
    },
    closeMenu(useTimeout) {
      if (this.disabled) {
        return
      }
      this.clearTimeouts()
      if (useTimeout === true) {
        this.popoverMouseLeave()
      } else {
        this.isPopoverOpen = false
      }
    },
    popoverMouseEnter() {
      if (this.disabled) {
        return
      }
      this.clearTimeouts()
      if (!this.isPopoverOpen) {
        mouseEnterTimer = setTimeout(() => {
          this.isPopoverOpen = true
        }, 500)
      }
    },
    popoverMouseLeave() {
      if (this.developerNoAutoClosing) return
      if (this.disabled) {
        return
      }
      this.clearTimeouts()
      if (this.isPopoverOpen) {
        mouseLeaveTimer = setTimeout(() => {
          this.isPopoverOpen = false
        }, 300)
      }
    },
    clearTimeouts() {
      clearTimeout(mouseEnterTimer)
      clearTimeout(mouseLeaveTimer)
    },
  },
}
</script>
