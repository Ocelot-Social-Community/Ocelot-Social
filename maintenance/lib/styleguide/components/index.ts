import Avatar from './data-display/Avatar/Avatar.vue'
import CopyField from './data-display/CopyField/CopyField.vue'
import List from './data-display/List/List.vue'
import Number from './data-display/Number/Number.vue'
import Table from './data-display/Table/Table.vue'
import Form from './data-input/Form/Form.vue'
import FormItem from './data-input/FormItem/FormItem.vue'
import Input from './data-input/Input/Input.vue'
import Radio from './data-input/Radio/Radio.vue'
import Select from './data-input/Select/Select.vue'
import Container from './layout/Container/Container.vue'
import Card from './layout/Card/Card.vue'
import Grid from './layout/Grid/Grid.vue'
import Modal from './layout/Modal/Modal.vue'
import Page from './layout/Page/Page.vue'
import PageTitle from './layout/PageTitle/PageTitle.vue'
import Placeholder from './layout/Placeholder/Placeholder.vue'
import Section from './layout/Section/Section.vue'
import Spinner from './layout/Spinner/Spinner.vue'
import Button from './navigation/Button/Button.vue'
import Menu from './navigation/Menu/Menu.vue'
import Chip from './typography/Chip/Chip.vue'
import Code from './typography/Code/Code.vue'
import Heading from './typography/Heading/Heading.vue'
import Icon from './typography/Icon/Icon.vue'
import Logo from './typography/Logo/Logo.vue'
import Tag from './typography/Tag/Tag.vue'
import Text from './typography/Text/Text.vue'

export {
  Avatar,
  CopyField,
  List,
  Number,
  Table,
  Form,
  FormItem,
  Input,
  Radio,
  Select,
  Container,
  Card,
  Grid,
  Modal,
  Page,
  PageTitle,
  Placeholder,
  Section,
  Spinner,
  Button,
  Menu,
  Chip,
  Code,
  Heading,
  Icon,
  Logo,
  Tag,
  Text,
}
