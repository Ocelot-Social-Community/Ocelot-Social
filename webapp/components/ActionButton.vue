<template>
  <div class="action-button">
    <base-button
      :loading="loading"
      :disabled="disabled"
      :filled="filled"
      :icon="icon"
      :aria-label="text"
      circle
      @click="click"
    />
    <div class="count">{{ count }}</div>
  </div>
</template>

<script>
export default {
  props: {
    count: { type: Number, required: true },
    text: { type: String, required: true },
    disabled: { type: Boolean, default: false },
    filled: { type: Boolean, required: true },
    icon: { type: String, required: true },
  },
  methods: {
    click() {
      this.$emit('click')
    },
  },
}
</script>

<style lang="scss" scoped>
.action-button {
  display: flex;
  justify-content: center;
  align-items: center;
  gap: $space-xx-small;
  position: relative;
  --icon-size: calc(var(--circle-button-width, #{$size-button-base}) / 2);
}
.count {
  user-select: none;
  color: $color-secondary-inverse;
  background-color: $color-primary-dark;
  display: flex;
  align-items: center;
  justify-content: center;
  position: absolute;
  top: -8px;
  right: -8px;
  --diameter: calc(var(--circle-button-width, #{$size-button-base}) * 0.7);
  width: var(--diameter);
  height: var(--diameter);
  border-radius: 12px;
  font-size: 12px;
}
</style>
