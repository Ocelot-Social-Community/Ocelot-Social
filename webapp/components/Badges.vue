<template>
  <div :class="[badges.length === 2 && 'hc-badges-dual']" class="hc-badges">
    <div v-for="badge in badges" :key="badge.id" class="hc-badge-container">
      <img :title="badge.key" :src="badge.icon | proxyApiUrl" class="hc-badge" />
    </div>
  </div>
</template>

<script>
export default {
  props: {
    badges: {
      type: Array,
      default: () => [],
    },
  },
}
</script>

<style lang="scss">
.hc-badges {
  text-align: center;
  position: relative;

  .hc-badge-container {
    display: inline-block;
    position: unset;
    overflow: hidden;
    vertical-align: middle;
  }

  .hc-badge {
    display: block;
    width: 100%;
  }

  $size: 30px;
  $offset: $size * -0.2;

  &.hc-badges-dual {
    padding-top: $size / 2 - 2;
  }

  .hc-badge-container {
    width: $size;
    height: 26px;
    margin-left: -1px;

    &:nth-child(3n - 1) {
      margin-top: -$size - $offset - 3;
      margin-left: -$size * 0.33 - $offset - 2;
    }
    &:nth-child(3n + 0) {
      margin-top: $size + $offset + 3;
      margin-left: -$size;
    }
    &:nth-child(3n + 1) {
      margin-left: -6px;
    }
    &:first-child {
      margin-left: -$size / 3;
    }
    &:last-child {
      margin-right: -$size / 3;
    }
  }
}
</style>
