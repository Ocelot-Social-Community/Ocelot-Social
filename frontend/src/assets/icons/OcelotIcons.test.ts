import { mount } from '@vue/test-utils'
import { navigate } from 'vike/client/router'
import { describe, it, expect, beforeEach, vi } from 'vitest'
import { VIcon } from 'vuetify/components'

vi.mock('vike/client/router')
vi.mocked(navigate).mockResolvedValue()

describe('Ocelot Icons', () => {
  const Wrapper = () => {
    return mount(VIcon, {
      props: {
        icon: '$history',
        color: 'green',
        size: 'large',
      },
    })
  }
  let wrapper: ReturnType<typeof Wrapper>

  beforeEach(() => {
    wrapper = Wrapper()
  })

  it('renders and tests if the ocelot icon "$history" is installed', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
