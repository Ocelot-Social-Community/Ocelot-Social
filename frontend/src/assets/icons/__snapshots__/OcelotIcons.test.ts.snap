// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Ocelot Icons > renders and tests if the ocelot icon "$history" is installed 1`] = `
<i
  aria-hidden="true"
  class="v-icon notranslate v-theme--light v-icon--size-large text-green"
>
  
  <!-- Generated by IcoMoon.io -->
  <svg
    fill="currentColor"
    height="32"
    stroke="currentColor"
    version="1.1"
    viewBox="0 0 32 32"
    width="32"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      history
    </title>
    <path
      d="M16 4c6.616 0 12 5.385 12 12s-5.385 12-12 12-12-5.385-12-12h2c0 5.535 4.465 10 10 10s10-4.465 10-10-4.465-10-10-10c-3.702 0-6.897 2.020-8.625 5h3.625v2h-7v-7h2v3.344c2.153-3.232 5.833-5.344 10-5.344zM15 8h2v7h5v2h-7v-9z"
    />
  </svg>
  
</i>
`;
